### PR TITLE
Take 2 of gittuf verification functions

### PR DESCRIPTION
**Pull request by adityasaky**
The last part of #52.

### CODE REVIEW

**Comment by adityasaky** [comment]
Addressed outstanding comments from #52.

Still relevant: https://github.com/adityasaky/gittuf/pull/52#discussion_r1261640641 and https://github.com/adityasaky/gittuf/pull/52#discussion_r1261638135

**Comment by adityasaky** [comment]
Changed from #52: this PR updates the CLI to gittuf verify-ref. This can be used to demonstrate branch protection features but file access rules are WIP while we work out the nuances of file rules in combination with git ref rules.

**Comment by adityasaky** [comment]
@wlynch I think this may be ready for another pass!

**Review by wlynch — COMMENTED** [review]
Looks great! Just one small thing.

File: internal/policy/verify.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+	// 1. Load policy applicable at firstEntry
+	state, err := LoadStateForEntry(ctx, repo, initialPolicyEntry)
+	if err != nil {
+		return err
+	}
+	currentPolicy = state
+
+	// 2. Enumerate RSL entries between firstEntry and lastEntry, ignoring irrelevant ones
+	iteratorEntry := lastEntry
+	for {
```
[the remark this answers is not in the document]
Comment (adityasaky) [reply internal/policy/verify.go:76]: In normal functioning that shouldn't happen. That said, this function walks back from the last entry to the first. If the first isn't found, it'll eventually walk all the way back to the start of the RSL at which point `GetParent` errors out.

[the remark this answers is not in the document]
Comment (wlynch) [reply internal/policy/verify.go:76]: Perfect - thanks!

**Review by wlynch — APPROVED** [review]

**Review by wlynch — APPROVED** [review]

**Pull request merged**